reports/report-manager.js

```diff
@@ -0,0 +1,226 @@
+var reportModels = require('./report-models');
+var Report = reportModels.Report;
+var EventReport = reportModels.EventReport;
+var OtherReport = reportModels.OtherReport;
+
+/**
+ * Callback used by report CRUD methods that return only one report.
+ * @callback reportCallback
+ * @param {Error} err - An error that occurred during the operation.  Null if no
+ *     errors occurred.
+ * @param {Report} report - The report operated on.
+ */
+
+/**
+ * Creates a report with the fields in the request object, and calls a callback
+ * when the database save finishes.
+ * @param {Object} reqData - The request object representing the report to be
+ *     created.
+ * @param {Object} locals - An object containing the current request's local
+ *     variables.
+ * @param {reportCallback} createCallback - Called once the operation finishes.
+ */
+var createReport = (reqData, locals, createCallback) => {
+  // TODO(jmtaber129): Check that the student UID matches the requesting user,
+  // or the requesting user is an admin.
+  var report;
+  if (reqData.type == 'event') {
+    report = new EventReport({
+      approvalStatus: false,
+      student: reqData.student,
+      event: reqData.event,
+    });
+  } else if (reqData.type == 'other') {
+    report = new OtherReport({
+      approvalStatus: false,
+      student: reqData.student,
+      category: reqData.category,
+      datetime: reqData.datetime,
```

Comment:
From what I've seen, mongoose implicitly converts date strings to Date objects, but I'll change it to explicitly convert them.